src/core/Akka.API.Tests/CoreAPISpec.ApproveCore.approved.txt

```diff
@@ -2540,45 +2523,47 @@ namespace Akka.Dispatch.MessageQueues
         public UnboundedMessageQueue() { }
         public int Count { get; }
         public bool HasMessages { get; }
-        public void Enqueue(Akka.Actor.Envelope envelope) { }
+        public void CleanUp(Akka.Actor.IActorRef owner, Akka.Dispatch.MessageQueues.IMessageQueue deadletters) { }
+        public void Enqueue(Akka.Actor.IActorRef receiver, Akka.Actor.Envelope envelope) { }
         public bool TryDequeue(out Akka.Actor.Envelope envelope) { }
     }
     public class UnboundedPriorityMessageQueue : Akka.Dispatch.MessageQueues.BlockingMessageQueue
     {
-        public UnboundedPriorityMessageQueue(System.Func<object, int> priorityGenerator) { }
+        public UnboundedPriorityMessageQueue(int initialCapacity) { }
+        public UnboundedPriorityMessageQueue(System.Func<object, int> priorityGenerator, int initialCapacity) { }
         protected override int LockedCount { get; }
         protected override void LockedEnqueue(Akka.Actor.Envelope envelope) { }
         protected override bool LockedTryDequeue(out Akka.Actor.Envelope envelope) { }
     }
 }
 namespace Akka.Dispatch.SysMsg
 {
-    public sealed class ActorTask : Akka.Actor.INoSerializationVerificationNeeded, Akka.Dispatch.SysMsg.ISystemMessage
+    public sealed class ActorTask : Akka.Dispatch.SysMsg.SystemMessage
     {
         public ActorTask(System.Threading.Tasks.Task task) { }
         public System.Threading.Tasks.Task Task { get; }
     }
-    public sealed class Create : Akka.Actor.INoSerializationVerificationNeeded, Akka.Dispatch.SysMsg.ISystemMessage
+    public sealed class Create : Akka.Dispatch.SysMsg.SystemMessage
     {
         public Create(Akka.Actor.ActorInitializationException failure = null) { }
         public Akka.Actor.ActorInitializationException Failure { get; }
         public override string ToString() { }
     }
-    public sealed class DeathWatchNotification : Akka.Actor.INoSerializationVerificationNeeded, Akka.Dispatch.SysMsg.ISystemMessage
+    public sealed class DeathWatchNotification : Akka.Dispatch.SysMsg.SystemMessage
     {
         public DeathWatchNotification(Akka.Actor.IActorRef actor, bool existenceConfirmed, bool addressTerminated) { }
         public Akka.Actor.IActorRef Actor { get; }
         public bool AddressTerminated { get; }
         public bool ExistenceConfirmed { get; }
         public override string ToString() { }
     }
-    public sealed class Escalate : Akka.Actor.INoSerializationVerificationNeeded, Akka.Dispatch.SysMsg.ISystemMessage
+    public sealed class Escalate : Akka.Dispatch.SysMsg.SystemMessage
     {
         public Escalate(System.Exception reason) { }
         public System.Exception Reason { get; }
         public override string ToString() { }
     }
-    public sealed class Failed : Akka.Actor.INoSerializationVerificationNeeded, Akka.Dispatch.SysMsg.ISystemMessage
+    public sealed class Failed : Akka.Dispatch.SysMsg.SystemMessage, Akka.Dispatch.SysMsg.IStashWhenFailed
```

Comment:
cc @rogeralsing one note; when we make an `async` / `await` call inside actors right now, SystemMessages are still allowed to be processed. I personally think that's fine - the only real thing that could cause an `ActorContext` to significantly change in the midst of that would be a child terminating, which we couldn't block anyway (since it happens inside the context of another actor.)

That being said, if we find a case where we need to suspend certain system messages during async await we can extend this `IStashWhen__` convention to support that.